src/app/register/register.component.html

```diff
@@ -0,0 +1,139 @@
+    >
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('lastName')?.touched && registrationForm.get('lastName')?.errors?.['required']"
+      >
+        *Last name is required
+      </small>
+
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('lastName')?.touched && registrationForm.get('lastName')?.errors?.['minlength']"
+      >
+        *Please enter at least 3 character
+      </small>
+    </app-input>
+
+    <app-input
+      placeholder="Enter your user name"
+      type="text"
+      label="User Name"
+      [control]="getControlName(registrationForm, 'userName')"
+    >
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('userName')?.touched && registrationForm.get('userName')?.errors?.['required']"
+      >
+        *User name is required
+      </small>
+
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('userName')?.touched && registrationForm.get('userName')?.errors?.['uniqueUserNameError']"
+      >
+        *This user is already registered
+      </small>
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('userName')?.touched && registrationForm.get('userName')?.errors?.['noSpaceAllowed']"
+      >
+        *No space allowed.
+      </small>
+    </app-input>
+
+    <app-input
+      placeholder="Enter your email"
+      type="email"
+      label="Email"
+      [control]="getControlName(registrationForm, 'email')"
+    >
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('email')?.touched && registrationForm.get('email')?.errors?.['required']"
+      >
+        *Email is required
+      </small>
+
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('email')?.touched && registrationForm.get('email')?.errors?.['email']"
+      >
+        *Enter a valid email
+      </small>
+
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('email')?.touched && registrationForm.get('email')?.errors?.['uniqueEmailError']"
+      >
+        *This email id is already taken
+      </small>
+    </app-input>
+    <app-input
+      placeholder="Enter your password"
+      type="password"
+      label="Password"
+      [control]="getControlName(registrationForm, 'password')"
+    >
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('password')?.touched && registrationForm.get('password')?.errors?.['required']"
+      >
+        *Password is required
+      </small>
+
+      <small
+        class="form__error-message"
+        *ngIf="registrationForm.get('password')?.touched && registrationForm.get('password')?.errors?.['minlength']"
+      >
+        *Please enter at least 6 character
+      </small>
+    </app-input>
+
+    <br /><br />
+
+    <app-button
+      [disabled]="registrationForm.invalid"
```

Comment:
We should follow the orders of attributes and/or bindings like below:

1. Structural directives (*ngFor, *ngIf, etc.)
2. Animation triggers (@fade, [@fade])
3. Element reference (#myComponent)
4. HTML attributes (class, etc.)
5. Non-interpolated string inputs and attributes (foo="bar")
6. Interpolated inputs ([bar]="theBar")
7. Two-way bindings ([(ngModel)]="email")
8. Outputs ((click)="onClick()")